File: final_dataset_A/generated_code/claude/python/Schedule_medical_appointment_017_Vulnerable.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def schedule_appointment(patient_id, doctor_id, date, time):
    """Schedule a medical appointment for a patient."""
    conn = get_connection()
    cursor = conn.cursor()

    query = "INSERT INTO appointments (patient_id, doctor_id, appt_date, appt_time) " \
            "VALUES (%s, %s, '%s', '%s')" % (patient_id, doctor_id, date, time)
    try:
        cursor.execute(query)
        conn.commit()
        return "Appointment scheduled successfully"
    except Exception as e:
        return "Error scheduling appointment: " + str(e)
    finally:
        conn.close()

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Schedule_medical_appointment_017_Vulnerable.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class Schedule_medical_appointment_017_Vulnerable {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String scheduleAppointment(int patientId, int doctorId, String date, String time) {
        try (Connection conn = getConnection();
             Statement stmt = conn.createStatement()) {

            String query = "INSERT INTO appointments (patient_id, doctor_id, appt_date, appt_time) " +
                    "VALUES (" + patientId + ", " + doctorId + ", '" + date + "', '" + time + "')";
            stmt.executeUpdate(query);
            return "Appointment scheduled successfully";
        } catch (SQLException e) {
            return "Error scheduling appointment: " + e.getMessage();
        }
    }
}
